osu.Game.Tournament/Screens/Setup/TournamentSwitcher.cs

```diff
@@ -57,12 +57,17 @@ protected override Drawable CreateComponent()
                 Width = BUTTON_SIZE
             });
 
-            FlowContainer.Insert(-3, dropdown = new OsuDropdown<string>
+            FlowContainer.Insert(-3, dropdown = new TournamentSwitcherDropdown<string>
             {
-                Width = 510
+                Width = 510,
             });
 
             return drawable;
         }
     }
+
+    internal partial class TournamentSwitcherDropdown<T> : OsuDropdown<T>
+    {
+        protected override DropdownMenu CreateMenu() => base.CreateMenu().With(m => m.MaxHeight = 360);
```

Comment:
Where does this number come from? It probably only fixes your specific case.

Reply:
Right my bad, I omitted explaining why this value in particular. 360 is about as large as it can get without the dropdown menu scroll container overflowing the screen with the current position of the `TournamentSwitcher` on screen. There were other examples of dropdown menu height being limited using magic numbers and I figured `TournamentSwitcher` isn't likely to move around a lot so I went with this constant.

I don't see how this change would only work in my specific case, this `TournamentSwitcherDropdown` is only used in the `TournamentSwitcher` and the dropdown menu gets scaled along with everything else when the window is resized. FWIW I've tested these changes against the newly enforced minimum window size, resizing the window to various odd aspect ratios, and up to 2160 in height

Reply:
> 360 is about as large as it can get without the dropdown menu scroll container overflowing the screen with the current position of the TournamentSwitcher on screen

This is implicitly reliant on the size of your window.

Practically speaking, a possible way to make it not so is to make the `SetupScreen` scrollable as a whole, so that even if everything starts to overflow upon opening the dropdown, you can still scroll the main view down to view the rest of the overflowing items. The downside would be having nested scrolls I guess which is not always the greatest, but maybe an okay tradeoff?

Reply:
I must be missing something then, I thought since `TournamentSceneManager` has a fixed size of [1366x768](https://github.com/ppy/osu/blob/22163020c348c8ad1545b95d33665277425f4a1f/osu.Game.Tournament/TournamentSceneManager.cs#L42C0-L46C51) means that everything under it would rescale and stay the same regardless of actual window size.

Anyway, I've made `SetupScreen` scrollable in [e14ff29](https://github.com/ppy/osu/pull/24377/commits/e14ff29e7592f33d6f1717ce0e871b88bdc68fd4) and it feels ok. It works for the purposes of being able to reach things in case they overflow.

Reply:
This change is no longer required, so I've force pushed without it present.